### PR TITLE
Add `settings --schema` support

### PR DESCRIPTION
**Pull request by cidrblock**
Generate a JSON schema for the ansible-navigator settings file.

(it defaults to `json`, which is the only choice today)

This is not 100% settings driven, the types and fields really need to wait until the settings entries are fully typed and annotation can be used to derive JSON types, but it is a start.

A future PR will use the json schema to validate the settings file the user has provided.

### CODE REVIEW

File: docs/changelog-fragments.d/1093.feature.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Added the ability to produce a json schema file for the settings file.
+
+```bash
+ansible-navigator settings --json-schema
```
Comment (ganeshrn) [inline docs/changelog-fragments.d/1093.feature.md:4]: How about `ansible-navigator settings --schema-type jsonschema`? Provides scope to add newer schema type if required.

File: src/ansible_navigator/actions/settings.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from dataclasses import asdict
 from typing import Tuple
 
+from ansible_navigator.configuration_subsystem.transform import to_schema
```
Comment (ganeshrn) [inline src/ansible_navigator/actions/settings.py:6]: Any specific reason to switch to absolute import?

Comment (cidrblock) [reply src/ansible_navigator/actions/settings.py:6]: no, just a miss, vscode auto-added it and I didn't correct it, TY

File: src/ansible_navigator/configuration_subsystem/navigator_configuration.py [hunk 3]
```diff
@@ -375,6 +375,15 @@ class Internals:
             subcommands=["inventory", "run"],
             value=SettingsEntryValue(),
         ),
+        SettingsEntry(
+            name="json_schema",
+            choices=[True, False],
+            cli_parameters=CliParameters(short="--js", action="store_true"),
+            settings_file_path_override="settings.json-schema",
+            short_description="Generate a json schema for the settings file",
```
Comment (ganeshrn) [inline src/ansible_navigator/configuration_subsystem/navigator_configuration.py:383]: Can also mention it generates draft7 schema

File: docs/changelog-fragments.d/1093.feature.md (hunk 1 above)
Comment (cidrblock) [reply docs/changelog-fragments.d/1093.feature.md]: done

File: src/ansible_navigator/configuration_subsystem/navigator_configuration.py (hunk 3 above)
Comment (cidrblock) [reply src/ansible_navigator/configuration_subsystem/navigator_configuration.py]: done

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**